src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -902,4 +905,43 @@ public String pollDeletedDetectorTask() {
         return this.deletedDetectorTasks.poll();
     }
 
+    /**
+     * Add deleted detector's id to deleted detector queue.
+     * @param detectorId detector id
+     */
+    public void addDeletedDetector(String detectorId) {
+        if (deletedDetectors.size() < maxCachedDeletedTask) {
+            deletedDetectors.add(detectorId);
+        }
+    }
```

Comment:
If the queue is full how do we handle an excess amount of detectors-to-be-deleted?

Reply:
`maxCachedDeletedTask` is 1000 by default (that means each node can have 1000 deleted detector in cache), should be enough to cache deleted detectors. If user delete too many detectors in short time, we will ignore excess deleted detectors, that means user may need to delete these detector's result manually with delete result API.